common/inc/BaseService.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Mbed-OS Microcontroller Library
+ * Copyright (c) 2020 Embedded Planet
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#ifndef MBED_OS_EXPERIMENTAL_BLE_SERVICES_INC_BASESERVICE_H_
+#define MBED_OS_EXPERIMENTAL_BLE_SERVICES_INC_BASESERVICE_H_
+
+/* TODO implement as decorator:
+ * - Way to add security requirements to characteristics (return list of characteristics?)
+ *
+ */
+
+//class BaseService
+//{
+//public:
+//
+//    BaseService();
+//
+//
+//
+//
+//};
+
+
+#endif /* MBED_OS_EXPERIMENTAL_BLE_SERVICES_INC_BASESERVICE_H_ */
```

Comment:
It's a good idea but perhaps we should add it in a separate PR with the other utility files:

* ConnectionThingy.h
* GattPresentationFormatDescriptor.h
* CharacteristicPresentationFormatDescriptor.h
* CharacteristicUserDescriptionDescriptor.h

Reply:
Ha, sorry, these files weren't supposed to be included in this PR :sweat_smile: